src/stenotype/cli.py

```diff
@@ -0,0 +1,82 @@
+"""Command Line Interface for stenotype.
+
+The backend is exposed here through a simple command line interface. Documentation of
+their usage is contained in the docstrings, which are written in a way that they will be
+rendered nicely by click's automatic --help text generator.
+"""
+from logging import getLogger
+
+import click
+
+from stenotype.backend import parser
+from stenotype import util
+
+log = getLogger(__name__)
+
+
+@click.command()
+@click.option(
+    "--loglevel",
+    default="INFO",
+    type=click.Choice(["DEBUG", "INFO", "WARNING", "ERROR"]),
+    help="Set the loglevel.",
+)
+@click.option(
+    "-f",
+    "--file",
+    type=click.File("r"),
+    help="Load a reference file if annotations need custom classes.",
+)
+@click.option(
+    "-s",
+    "--shorten",
+    is_flag=True,
+    help="Inverts the parser to instead try to generate stenotype from standard types.",
+)
+@click.option(
+    "-c",
+    "--check",
+    is_flag=True,
+    help="Disables the check-skipping. Annotations need to be written as they would "
+    "be in source files in this mode.",
+)
+@click.argument("args", nargs=-1)
+def cli(args, loglevel, file, shorten, check):
+    """CLI entrypoint to test the stenotype backend.
+
+    Enter any number of stenotype expressions to see which standard annotations they will
+    resolve into. There is no need to double quote expression or to assign them to a
+    value, the following will just work:
+
+    \b
+      $ stenotype 'bool or int'
+      typing.Union[bool, int]
+      $ stenotype '(int) -> bool'
+      typing.Signature[[int], bool]
+
+    You can also enter multiple arguments that will be executed in turn and consider each
+    other in the context of, for example, custom type variables:
+
+    \b
+      $ stenotype 'T, int' 'bool or T'
+      T = TypeVar[int]
+      typing.Union[bool, T]
+    """
+    util.setup_logging(loglevel)
+
+    # early exits
+    if shorten and check:
+        print("The flags --shorten and --check can not be used at the same time.")
+        exit(1)
+    if not args:
+        print("No arguments entered, nothing to do.")
+
+    # control behavior based on input parameters
+    context = None
+    if file:
+        context = compile(file.read(), filename="<string>", mode="exec")
+
+    if shorten:
+        parser.inverse_parse(args, context)
+    else:
+        parser.parse(args, context)
```

Comment:
The CLI should return a non-zero exit code if parsing does not work (see above comment on use by other tools). Consider defining an Exception for parsing failures, and having the CLI frontend transform that to a human-readable error message.

Perhaps just drop a ``# TODO`` and have this as a good-first-issue.

Reply:
These kind of Exceptions are usually just no-content `class StenotypeException(Exception): pass`, right?

I think I'll add that section already, it will be changed once the backend API s written anyway and some of the testing with `click` can be strange.

Reply:
> having the CLI frontend transform that to a human-readable error message

Right no it's just `click.echo(e, err=True)`, would you say `traceback.print_exc()` is better?

Reply:
We can go through various levels of error report quality -- from a simple ``SyntaxError: "A and B"`` to ``SyntaxError: unrecognised "and" in stenotype "A and B"`` to some [ELM/Rust like ASCII help printout](https://elm-lang.org/news/compiler-errors-for-humans).

So yeah, ``click.echo`` is a good start unless someone is really, really bored. ;)